pkg/util/dns_name.go

```diff
@@ -9,28 +9,22 @@ import (
 var regex = regexp.MustCompile(`[a-z0-9]`)
 
 // DNSName returns a dns-safe string for the given name.
-// Any char that is not [a-z0-9] is replaced by "-".
-// If the final name starts with "-", "a" is added as prefix. Similarly, if it ends with "-", "z" is added.
+// Any char that is not [a-z0-9] is replaced by "-" or "a".
+// Replacement character "a" is used only at the beginning or at the end of the name.
+// The function does not change length of the string.
 func DNSName(name string) string {
```

Comment:
I have changed the behavior, the issue was that the function was changing the length of the `name` and in our codebase the `DNSName` was applied after the truncation.

There is no need to use `a` and `z` as replacement chars. The original name cannot be reconstructed.